x-pack/plugins/ml/public/application/model_management/test_models/selected_model.tsx

```diff
@@ -22,52 +24,134 @@ import {
+  const pipeline = (createPipelineFlyoutOpen && pipelineConfig) || undefined;
 
   const inferrer = useMemo<InferrerType | undefined>(() => {
-    if (model.model_type === TRAINED_MODEL_TYPE.PYTORCH) {
-      const taskType = Object.keys(model.inference_config ?? {})[0];
+    const taskType = Object.keys(model.inference_config ?? {})[0];
+    let tempInferrer;
 
+    if (model.model_type === TRAINED_MODEL_TYPE.PYTORCH) {
       switch (taskType) {
         case SUPPORTED_PYTORCH_TASKS.NER:
-          return new NerInference(trainedModels, model, inputType, deploymentId);
+          tempInferrer = new NerInference(trainedModels, model, inputType, deploymentId);
+          break;
         case SUPPORTED_PYTORCH_TASKS.TEXT_CLASSIFICATION:
-          return new TextClassificationInference(trainedModels, model, inputType, deploymentId);
+          tempInferrer = new TextClassificationInference(
+            trainedModels,
+            model,
+            inputType,
+            deploymentId
+          );
+          break;
         case SUPPORTED_PYTORCH_TASKS.ZERO_SHOT_CLASSIFICATION:
-          return new ZeroShotClassificationInference(trainedModels, model, inputType, deploymentId);
+          tempInferrer = new ZeroShotClassificationInference(
+            trainedModels,
+            model,
+            inputType,
+            deploymentId
+          );
+          break;
         case SUPPORTED_PYTORCH_TASKS.TEXT_EMBEDDING:
-          return new TextEmbeddingInference(trainedModels, model, inputType, deploymentId);
+          tempInferrer = new TextEmbeddingInference(trainedModels, model, inputType, deploymentId);
+          break;
         case SUPPORTED_PYTORCH_TASKS.FILL_MASK:
-          return new FillMaskInference(trainedModels, model, inputType, deploymentId);
+          tempInferrer = new FillMaskInference(trainedModels, model, inputType, deploymentId);
+          break;
         case SUPPORTED_PYTORCH_TASKS.QUESTION_ANSWERING:
-          return new QuestionAnsweringInference(trainedModels, model, inputType, deploymentId);
+          tempInferrer = new QuestionAnsweringInference(
+            trainedModels,
+            model,
+            inputType,
+            deploymentId
+          );
+          break;
         case SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION:
-          return new TextExpansionInference(trainedModels, model, inputType, deploymentId);
+          tempInferrer = new TextExpansionInference(trainedModels, model, inputType, deploymentId);
+          break;
         default:
           break;
       }
     } else if (model.model_type === TRAINED_MODEL_TYPE.LANG_IDENT) {
-      return new LangIdentInference(trainedModels, model, inputType, deploymentId);
+      tempInferrer = new LangIdentInference(trainedModels, model, inputType, deploymentId);
     }
-  }, [inputType, model, trainedModels, deploymentId]);
+    if (tempInferrer) {
+      tempInferrer.setSwitchtoCreationMode(setCurrentContext);
```

Comment:
```suggestion
      tempInferrer.setSwitchtoCreationMode(setCurrentContext);    if (tempInferrer) {
      tempInferrer.setSwitchtoCreationMode(() => {
        if (tempInferrer) {
          setCurrentContext({
            pipelineConfig: tempInferrer.getPipeline(),
            defaultSelectedDataViewId: tempInferrer.getSelectedDataViewId(),
            createPipelineFlyoutOpen: true,
          });
        }
      });
    }
```

By setting it all in the callback it means you only need to call `inferrer.switchToCreationMode()` to run it